### PR TITLE
added arch? check including manjaro call

### PR DESCRIPTION
**Pull request by chayim**
This starts to add Arch Linux support for modules that support it. By implementing an function called _arch_compat_, Redis modules can add explicit Arch support.

### CODE REVIEW

**Comment by rafie** [comment]
Very nice.
Two important comments:
(-) Better not to use `sudo` in `install()` and `uninstall()`. There are several reasons for that:
    (-) No sudo in docker (unless installed),
    (-) If running as non-root, users will use `sudo system-install.py`,
    (-) sudo sometimes will cause scripts to fail (with double-invocation or when env vars are not passed through it). For sake of reliability, it's better to avoid it inside scripts.
(-) I consider `arch` as an archetype (much like Ubuntu). Like, systems that are arch-based give `dist == 'arch'` and `brand == 'manjero'` (same happens with `elementary OS` and `pop_OS`, which are identified as `ubuntu`). The reason for that is that we try to understand what can/cannot be installed on a particular OS, so it's makes more sense to identify an OS as its archetype. `debian_compat` and `redhat_compat` are way to group OSes that have some similarity (maybe same package manager), but still have noticeable difference. According to that, maybe `linuxmint` should identify as `ubuntu`.

**Comment by chayim** [comment]
I'm limiting the scope of these changes to arch support only, and found a simpler method, relying on id_like. The major arch distributions, including Arch (Manarjo, Arco, and BlackArch) all maintain id_like for Arch. 

We can scope a larger PR that revisits operating systems and issues like Ubuntu variants like _linuxmint_, _elementary_, etc.

File: paella/setup.py [hunk 1]
```diff
@@ -219,8 +219,17 @@ class Pacman(PackageManager):
     def __init__(self, runner):
         super(Pacman, self).__init__(runner)
 
-    def install(self, packs, group=False, output="on_error", _try=False):
-        return self.run("pacman --noconfirm -S " + packs, output=output, _try=_try, sudo=True)
+    def install(self, packs, group=False, output="on_error", _try=False, aur=False):
+        if aur is False:
```
Comment (rafie) [inline paella/setup.py:223]: `not aur`

File: paella/setup.py [hunk 2]
```diff
@@ -219,8 +219,17 @@ class Pacman(PackageManager):
...
+            return self.run("sudo pacman --noconfirm -S " + packs, output=output, _try=_try, sudo=True)
```
Comment (rafie) [inline paella/setup.py:224]: `sudo pacman` -> `pacman`

File: bin/getrmpytools [hunk 3]
```diff
@@ -44,6 +44,10 @@ class RMPyToolsSetup(paella.Setup):
         if not self.psutil_installed:
             self.build_and_install("python%s-devel" % self.pyver)
 
+    def arch_compat(self):
```
Comment (rafie) [inline bin/getrmpytools:47]: `def archlinux(self):`

File: paella/setup.py (hunk 1 above)
Comment (chayim) [reply paella/setup.py:223]: Why _not aur_? It's more python to explicitly check on is False (like is None).

Comment (rafie) [reply paella/setup.py:223]: Right. I assumed `aur` is boolean, and I always opt for the more compact syntax. But you're right.

**Review by rafie — APPROVED** [review]

**Pull request merged**